src/longbow/segment/command.py

```diff
@@ -594,21 +608,29 @@ def _write_split_array_element(
     if barcode_conf_file is not None and a.has_tag(bam_utils.READ_BARCODE_CONF_FACTOR_TAG):
         barcode_conf_file.write(f"{a.get_tag(bam_utils.READ_RAW_BARCODE_TAG)}\t{a.get_tag(bam_utils.READ_BARCODE_CONF_FACTOR_TAG)}\n")
 
-    bam_out.write(a)
+    has_cbc_and_umi = bam_utils.has_cbc_and_umi(a)
```

Comment:
In the future I think we should also cue off the model itself and disable this check if the model doesn't have a CBC / UMI field.

Reply:
Agreed